### ISSUE TITLE
Error decoding webhook data (eof) seen again

### ISSUE THREAD

**Issue by a-roberts** [tektoncd/experimental#291]
# Expected Behavior

A PipelineRun for a webhook should be kicked off successfully

# Actual Behavior

```
Error from server (BadRequest): a container name must be specified for pod webhooks-extension-sink-mtzqt-deployment-5d769d468f-gb5f9, choose one of: [user-container queue-proxy]
[root@requite1 example-pipelines]# k logs webhooks-extension-sink-mtzqt-deployment-5d769d468f-gb5f9 -f -c user-container
{"level":"info","msg":"constructed a logger"}
{"level":"info","msg":"constructed a logger"}
{"level":"info","msg":"Creating server and entering wait loop."}
{"level":"info","msg":"In HandleWebhook for a GitHub event..."}
{"level":"info","msg":"Github event name to look for is: Ce-Github-Event."}
{"level":"debug","msg":"GitHub event type is: push."}
{"level":"info","msg":"Handling a push event."}
{"level":"error","msg":"error decoding webhook data: EOF."}
```

Carlos previously fixed an instance of this https://github.com/tektoncd/experimental/pull/286 which was to address a bug introduced by https://github.com/tektoncd/experimental/pull/264

Important code: https://github.com/tektoncd/experimental/blob/master/webhooks-extension/pkg/endpoints/sink.go

# Steps to Reproduce the Problem

1. Install the webhooks extension: I used the latest binary (our "nightly" which is actually the latest built code) on OpenShift
2. Create a webhook
3. I'm running simple-pipeline-insecure with a repo on github.ibm.com
4. Observe you get a webhooks extension sink pod come up and no PipelineRun due to the above error

**Comment by a-roberts** [comment]
Note that the data's read in fine using the latest code from master on Docker Desktop and simple pipeline, so it may be to do with simple pipeline insecure or the latest nightly images aren't actually up to date/containing fixes.

**Comment by vtereso** [comment]
I installed the dashboard and experimental using the latest release as well as their nightly releases and I could not reproduce this problem. I tested on the latest D4D and for the dependencies, I just ran the prereq script (https://github.com/tektoncd/experimental/blob/master/webhooks-extension/test/install_prereqs.sh). The only issue I could find was the monitor test was using the old interpolation syntax, which I opened https://github.com/tektoncd/experimental/pull/298 for.
